types/appconfig.pp

```diff
@@ -0,0 +1,24 @@
+type Puppetwebhook::Appconfig = Struct[{
+  protected                 => Boolean,
+  user                      => String[1],
+  pass                      => String[1],
+  client_cfg                => Stdlib::Absolutepath,
+  client_timeout            => String[1],
+  use_mco_ruby              => Boolean,
+  use_mcollective           => Boolean,
+  discovery_timeout         => String[1],
+  chatops                   => Boolean,
+  Optional[chatops_service] => Enum['slack', 'rocketchat'],
+  Optional[chatops_channel] => Pattern[/\A#.*/],
+  Optional[chatops_user]    => String[1],
+  Optional[chatops_url]     => Stdlib::HTTPUrl,
+  default_branch            => String[1],
+  ignore_environments       => Array[String[1]],
+  prefix_command            => String[1],
+  r10k_deploy_arguments     => String[1],
+  allow_uppercase           => Boolean,
+  command_prefix            => String[1],
+  prefix                    => String[1],
+  github_secret             => String[1],
+  repository_events         => Array,
```

Comment:
Do we know the datatypes for this array? Otherwise I'm fine with the PR.

Reply:
I don't think I know exactly what repository_events are or look like, so I don't want to restrict what goes into the Array until I know